commons/src/main/java/org/infinispan/commons/dataconversion/ByteArrayWrapper.java

```diff
@@ -0,0 +1,27 @@
+package org.infinispan.commons.dataconversion;
+
+import org.infinispan.commons.marshall.WrappedByteArray;
+
+/**
+ * Wraps byte[] on a {@link WrappedByteArray} to provide equality and hashCode support, leaving other objects
+ * unchanged.
+ *
+ * @since 9.1
+ */
+public class ByteArrayWrapper implements Wrapper {
+
```

Comment:
There are some more such singletons introduced in this PR. Should we also mark them final?